src/Data/YAML/Dumper.hs

```diff
@@ -40,49 +40,68 @@ dumpEvents nodes = Right StreamStart: go0 nodes
+      SFloat double -> Right (YE.Scalar anc tagFloat Plain (dumpFloat double))
+      SInt   int    -> Right (YE.Scalar anc tagInt   Plain (T.pack . show $ int))
+      SStr   text   -> Right (YE.Scalar anc untagged DoubleQuoted text)  -- or TODO: modify {- goStr text -}
+      SUnknown _ _  -> Left "SUnknown Scalar type"
+      -- where 
+      --   goStr :: T.Text -> Either String Event
+      --   goStr t
+      --     | T.null t          = Right (YE.Scalar anc tagStr Plain t)
+      --     | hasLeadSpace t    = if T.last t == '\n' then Right (YE.Scalar anc tagStr (Literal Keep IndentOfs2) t) else Right (YE.Scalar anc tagStr (Literal Clip IndentOfs2) t)
+      --     | T.last t == '\n'  = Right (YE.Scalar anc tagStr (Literal Keep IndentOfs2) t)
+      --     | otherwise         = Right (YE.Scalar anc tagStr Plain t)
+      --         where
+      --           hasLeadSpace t' = T.isPrefixOf " " . T.dropWhile (== '\n') $ t'
+
+    goMap :: Int -> Mapping () -> [Node ()] -> Node2EvList -> EvList
+    goMap lvl m rest cont = goNode lvl (mapToList m) g
+      where 
+        g []    = (Right MappingEnd) : isDocEnd (lvl - 1) rest cont
+        g rest' = goNode lvl rest' g 
+        mapToList = foldrWithKey (\k v a -> k : v : a) []
 
     goSeq :: Int -> [Node ()] -> [Node ()] -> Node2EvList -> EvList
-    goSeq lvl nod rest cont =  goNode lvl nod g
-        where 
-          g [] = (Right SequenceEnd) : isDocEnd (lvl - 1) rest cont
-          g rest' = goNode lvl rest' g 
+    goSeq lvl nod rest cont = goNode lvl nod g
+      where 
+        g []    = (Right SequenceEnd) : isDocEnd (lvl - 1) rest cont
+        g rest' = goNode lvl rest' g 
 
     goAnchor :: Int -> NodeId -> Node () -> [Node ()] -> Node2EvList -> EvList
     goAnchor lvl nid nod rest cont = case nod of 
-        YI.Scalar _ scalar -> goScalar scalar (ancName nid): isDocEnd lvl rest cont
-        Mapping   _ tag m  -> Right (MappingStart (ancName nid) tag Block) : goMap (lvl+1) m rest cont
-        Sequence  _ tag s  -> Right (SequenceStart (ancName nid) tag Block) : goSeq (lvl+1) s rest cont
-        Anchor    _ _ _    -> Left "Anchor has a anchor node" : (cont rest)
+      YI.Scalar _ scalar -> goScalar scalar (ancName nid): isDocEnd lvl rest cont
+      Mapping   _ tag m  -> Right (MappingStart (ancName nid) tag Block) : goMap (lvl + 1) m rest cont
+      Sequence  _ tag s  -> Right (SequenceStart (ancName nid) tag Block) : goSeq (lvl + 1) s rest cont
+      Anchor    _ _ _    -> Left "Anchor has a anchor node" : (cont rest)
 
     isDocEnd :: Int -> [Node ()] -> Node2EvList -> EvList
     isDocEnd lvl rest cont = if lvl == 0 then Right (DocumentEnd (rest /= [])): (cont rest) else (cont rest)
 
     ancName :: NodeId -> Maybe Anchor
     ancName (-1) = Nothing
     ancName nid  = Just $ T.pack ("a" ++ show nid)
+
+    dumpBool :: Bool -> T.Text
+    dumpBool b = if b then "true" else "false"
+
+    dumpFloat :: Double -> T.Text
```

Comment:
`SFloat` is converted to `Text` after making this assumption in [JSON schema](https://github.com/yaml/yaml-spec/issues/29)